api/src/task/xml_generation/generate_xml_cli.py

```diff
@@ -0,0 +1,116 @@
+"""CLI command for generating application XML components (header, footer, etc.)."""
+
+import logging
+import uuid
+
+import click
+
+import src.adapters.db as db
+from src.adapters.db import flask_db
+from src.services.applications.get_application import get_application
+from src.services.xml_generation.header_generator import generate_application_header_xml
+from src.task.task_blueprint import task_blueprint
+from src.util.local import error_if_not_local
+
+logger = logging.getLogger(__name__)
+
+
+@task_blueprint.cli.command(
+    "generate-submission-xml",
+    help="Generate submission XML components (header, footer, etc.) for a grant application",
+)
+@click.option(
+    "--application-id",
+    required=True,
+    help="UUID of the application to generate XML for",
+)
+@click.option(
+    "--part",
+    type=click.Choice(["header", "footer", "both"], case_sensitive=False),
+    default="header",
+    help="Which part of the XML to generate (default: header)",
+)
+@click.option(
+    "--pretty/--no-pretty",
+    default=True,
+    help="Pretty print the XML output (default: True)",
+)
+@click.option(
+    "--output-file",
+    help="Optional file path to write the XML to (prints to stdout if not specified)",
+)
+@flask_db.with_db_session()
+def generate_xml(
```

Comment:
Fine if we keep it if it helps with building it, but the only place we'll be using this is in our greater XML generation logic which'll only ever get called when we create the application zip.